### PR TITLE
Fix mac hc call problem

### PR DESCRIPTION
**Pull request by freesig**
Fixes some of #713

### CODE REVIEW

File: crates/hc_sandbox/src/calls.rs [hunk 1]
```diff
@@ -224,7 +224,9 @@ pub async fn call(holochain_path: &Path, req: Call) -> anyhow::Result<()> {
                 Ok(cmd) => cmds.push((cmd, None)),
                 Err(e) => {
                     if let holochain_websocket::WebsocketError::Io(e) = &e {
-                        if let std::io::ErrorKind::ConnectionRefused = e.kind() {
+                        if let std::io::ErrorKind::ConnectionRefused
+                        | std::io::ErrorKind::AddrNotAvailable = e.kind()
+                        {
```
Comment (steveeJ) [inline crates/hc_sandbox/src/calls.rs:229]:
```suggestion
                        if let
                        #[cfg(not(target_os = "macos"))]
                        std::io::ErrorKind::ConnectionRefused
                        #[cfg(target_os = "macos")]
                        std::io::ErrorKind::AddrNotAvailable = e.kind()
                        {
```

**Review by steveeJ — COMMENTED** [review]
It would be nice to see in the code that there are different errors by platform.
I added a suggestion but I'm not confident this compiles. A comment would probably suffice here :smile:

File: crates/hc_sandbox/src/calls.rs (hunk 1 above)
Comment (steveeJ) [reply crates/hc_sandbox/src/calls.rs:229]: @freesig ping

Comment (freesig) [reply crates/hc_sandbox/src/calls.rs:229]: Is it not better to just cover both cases?

Comment (steveeJ) [reply crates/hc_sandbox/src/calls.rs:226]: my suggestion is based on the idea that if one of the errors happen on the platform we didn't expect them on the treatment might be different. if that's not the case then my suggestion can be ditched

**Review by neonphog — APPROVED** [review]
I slightly prefer `*not*` doing the os targeting.

**Comment by steveeJ** [comment]
> I slightly prefer `*not*` doing the os targeting.

that's fine with me as well :wink:

**Review by steveeJ — APPROVED** [review]

**Pull request merged**